tools/pipelines-tasks/azure-pipelines/build-vsix.yml

```diff
@@ -23,93 +20,121 @@ variables:
+    ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    sdl:
+      git:
+        submodules: disable
+      spotBugs:        
+        enabled: false
+      baseline:
+        baselineFile: $(Build.SourcesDirectory)\tools\pipelines-tasks\azure-pipelines\.gdnbaselines
+      suppression:
+        suppressionFile: $(Build.SourcesDirectory)\tools\pipelines-tasks\azure-pipelines\.gdnsuppress
 
-- task: TfxInstaller@3
-  displayName: 'Use Node CLI for Azure DevOps (tfx-cli): v0.7.x'
+    pool:
+      name: Azure-Pipelines-1ESPT-ExDShared
+      image: windows-2019
+      os: windows
 
-# Package and sign the VSIX
-- task: PackageAzureDevOpsExtension@3
-  displayName: 'Package Extension: $(tasksRoot)'
-  inputs:
-    rootFolder: '$(tasksRoot)'
-    patternManifest: 'vss-extension.json'
-    outputPath: '$(buildOutRoot)\MsixPackagingExtension.vsix'
-    publisherId: 'MSIX'
-    extensionVersion: '$(major).$(minor).$(patch)'
-    extensionVisibility: public
-    updateTasksVersion: true
+    customBuildTags:
+    - ES365AIMigrationTooling
+    
+    stages:
+    - stage: stage
+      jobs:
+      - job: job
+        templateContext:  
+          outputs:
+          - output: pipelineArtifact
+            displayName: 'Publish VSIX artifact'
+            targetPath: '$(buildOutRoot)\MsixPackagingExtension.vsix'
+            artifactName: 'VSIX'
+            publishLocation: 'pipeline'
+            sbomBuildDropPath:  $(buildOutRoot)
 
-# Generate and publish SBoM file
-# It is placed under a _manifest folder by default
-- task: ManifestGeneratorTask@0
-  displayName: 'Generate Software Bill of Material (SBoM)'
-  inputs:
-      BuildDropPath: $(buildOutRoot)
-      BuildComponentPath: '$(tasksRoot)'
-      AdditionalComponentDetectorArgs: '--DirectoryExclusionList $(tasksRoot)\test'
-      PackageName: 'MSIX Packaging Extension'
-      PackageVersion: '$(major).$(minor).$(patch)'
+          # Publish privately
+          # Use different task IDs from public extension to prevent clashing.
+          - output: adoExtension
+            displayName: 'Publish Extension'
+            targetPath: $(buildOutRoot)
+            connectedServiceName: 'Visual Studio Marketplace - MSIX'
+            fileType: vsix
+            vsixFile: '$(buildOutRoot)\MsixPackagingExtension.vsix'
+            extensionId: 'msix-ci-automation-task-dev'
+            extensionName: 'MSIX Packaging (Preview)'
+            extensionVersion: '$(major).$(minor).$(patch)'
+            updateTasksId: true
+            extensionVisibility: 'privatepreview'
+            sbomBuildDropPath:  $(buildOutRoot)
 
-- task: DropValidatorTask@0
-  inputs:
-    BuildDropPath: $(buildOutRoot)
-    OutputPath: $(buildOutRoot)/_manifest/sbom_validation.json
-    ValidateSignature: true
+        steps:
+        - template: /tools/pipelines-tasks/azure-pipelines/templates/build-steps.yml@self
```

Comment:
Can this be just templates/build-steps.yml

Reply:
The path is now relative to the root of the repo. all the steps of build-steps.yml are also running successfully in the pipeline.